applications/openrollup-mvp-phase-1.md

```diff
@@ -0,0 +1,179 @@
+
+There are many zk-rollup projects, such as zksync, starknet, hermez, plonky2, they are basically in the Ethereum ecosystem. They are all operated by a central operator, and in our protocol, each zkapp is an operator.
+
+
+## Team :busts_in_silhouette:
+
+### Team members
+
+- Chris Cen (Lead developer)
+- MJewzk (Developer)
+
+### Contact
+
+- **Contact Name:** Chris Cen
+- **Contact Email:** chris@icen.cc
+- **Website:** https://openrollup.org/ (not yet established)
+
+### Legal Structure
+
+- **Registered Address:**  not yet available
+- **Registered Legal Entity:** not yet available
+
+### Team's experience
+
+Our team  have much experience in applied cryptography, blockchain research, and software development. 
+Chris has been a software engineer for over many years in the gaming industry and has led the development of various multiplayer online games. Recently he develop a hybrid consensus of Progpow and AuthorityRound [https://github.com/chrisicen/openethereum-unity](https://github.com/chrisicen/openethereum-unity). He is currently devoted to the development of off-chain ZK scalability solution for blockchain.
+MJewzk is a blockchain development engineer and has been involved in the development of some blockchain and web3 projects.
+
+### Team Code Repos
+
+- https://github.com/openrollup-zk
+- https://github.com/chrisicen
+- https://github.com/MJewzk
+
+### Team LinkedIn Profiles (if available)
+
+not yet available
+
+## Development Status :open_book:
+
+We've been working on techniques for zk-rollup, and verifiable computation for some time, including projects like zksync, halo2, plonky2, winterfell, hermez, and related papers and articles. We are starting to implement the rollup part.
+
+## Development Roadmap :nut_and_bolt:
+
+### Overview
+
+- **Total Estimated Duration:** 2.7 months
+- **Full-Time Equivalent (FTE):** 1.5 FTE
+- **Total Costs:** 48,600 USD).
+
+### Milestone 1 — Implement Substrate Modules
+
+- **Estimated duration:** 2.7 month
+- **FTE:**  1.5
+- **Costs:** 48,600 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | Apache 2.0 |
+| 0b. | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can spin up one of our Substrate nodes and send test transactions, which will show how the new functionality works. |
+| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an **article**/workshop that explains what was done/achieved as part of the grant. |
+| 1.  | Setup | We provide a repository that forms the basis of our deliverables.|
+| 2.  | Core Types | We provide the core data types, as specified in Project Details. |
```

Comment:
So these are the data structures, that will be used in your pallet? If so, I think it's sufficient to integrate it into the pallet deliverable. If there is a specific reason to do it in a separate fashion, e.g. in a dedicated library, pls elaborate on the reasons why you'd want to do it this way.

Reply:
✅